TemporaryParser-Tests/TPRBParserTest.class.st

```diff
@@ -168,104 +178,437 @@ TPRBParserTest >> testBlockReturnNode [
+  1; node.
+  1; node'.
+]
+
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageForgottenDotAfterAssignement [
+	| tree child error |
+	self shouldnt: [self parserClass parseExpression: '|temp | temp := 1 + 2 temp + 3'] raise: SyntaxErrorNotification.
+	
+	tree := self parserClass parseExpression: '|temp | temp := 1 + 2 temp + 3'.
+	self assert: tree isSequence.
+	
+	"There should only be one statement as there is no dot."
+	child := tree statements.
+	self assert: (child size) equals: 1.
+	self assert: child first isAssignment.
+	
+	"Getting the part changed by missing dot."
+	error := child first value receiver arguments first.
+	self assert: error isMessage.
+	self assert: error receiver value equals: 2.
+	self assert: error selector equals: #temp
+	
+	
+	
+]
+
+{ #category : #'garbage tests' }
+TPRBParserTest >> testGarbageForgottenReceiver [
+	"Test made in sight of implementing the search of previous litterals when encountering an unknown 
+	 variable in case it is, in fact, a selector. This works for cascade (forgotten ;) or missing 
+	 receiver like self ..."
+	self should: [self parserClass parseExpression: '|temp | temp := 1 + 2. + 3'] raise: SyntaxErrorNotification.
+]
+
+{ #category : #testIntervals }
+TPRBParserTest >> testGeneralBlockIntervalsArray [
+	"This tests that array nodes inside a block node correspond the right interval in the source code."
+	| tree str |
+	tree := (self parseExpression: '["comment node" | temporaries | message node. #literalNode . #(literal array node). {array node}. [block node]. cascade node; cascaded. assignement := node. ^return node]').
+	str := '{array node}'.
+	self assert: ((tree body statements at: 4) start) equals: (tree source findString: str).
+	self assert: ((tree body statements at: 4) stop) equals: ((tree source findString: str) + (str size - 1)).
+]
+
+{ #category : #testIntervals }
+TPRBParserTest >> testGeneralBlockIntervalsAssignement [
+	"This tests that assignement nodes inside a block node correspond the right interval in the source code."
+	| tree str |
+	tree := (self parseExpression: '["comment node" | temporaries | message node. #literalNode . #(literal array node). {array node}. [block node]. cascade node; cascaded. assignement := node. ^return node]').
+	str := 'assignement := node'.
+	self assert: ((tree body statements at: 7) start) equals: (tree source findString: str).
+	self assert: ((tree body statements at: 7) stop) equals: ((tree source findString: str) + (str size - 1)).
+]
+
+{ #category : #testIntervals }
+TPRBParserTest >> testGeneralBlockIntervalsBlock [
+	"This tests that block nodes inside a block node correspond the right interval in the source code."
+	| tree str |
+	tree := (self parseExpression: '["comment node" | temporaries | message node. #literalNode . #(literal array node). {array node}. [block node]. cascade node; cascaded. assignement := node. ^return node]').
+	str := '[block node]'.

```

Comment:
Also, a general thing you could do is to insert the code to avoid having it repeated twice:

```smalltalk
str := '[block node]'.
tree := (self parseExpression: '[
    "comment node"
    | temporaries |
    message node.
    #literalNode .
    #(literal array node).
    ', str, '.
    [block node].
    cascade node;
    cascaded.
    assignement := node.
    ^return node ]').
```